### PR TITLE
Logging change

### PR DESCRIPTION
**Pull request by isaac-tfn**
Describe Changes
---

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `setup.cfg`

### CODE REVIEW

File: src/caf/space/__main__.py [hunk 1]
```diff
@@ -1,25 +1,36 @@
...
     else:
-        config = caf.space.ZoningTranslationInputs.load_yaml(args.config_path)
-        trans = caf.space.ZoneTranslation(config)
-        if args.mode == "spatial":
-            out = trans.spatial_translation().to_csv(
-                args.out_path / f"{config.zone_1.name}_{config.zone_2.name}_spatial.csv",
-                ignore_index=True,
-            )
+        log_out = Path(os.getcwd())
+    with LogHelper("SPACE", details, log_file=log_out / "SPACE.log"):
```
Comment (wsp-mbuckley) [inline src/caf/space/__main__.py:18]: Use `__package__` for the root logger (which will be 'caf.space').

```suggestion
    with LogHelper(__package__, details, log_file=log_out / "SPACE.log"):
```

File: src/caf/space/ui.py [hunk 2]
```diff
@@ -771,7 +770,7 @@ class NotebookApp(tk.Tk):
 
     def __init__(self):
         super().__init__()
-        # self.logger = logging.getLogger("SPACE")
+        self.logger = logging.getLogger("SPACE")
```
Comment (wsp-mbuckley) [inline src/caf/space/ui.py:773]: Use `__package__` which will be 'caf.space'

```suggestion
        self.logger = logging.getLogger(__package__)
```

File: src/caf/space/ui.py [hunk 3]
```diff
@@ -798,38 +797,38 @@ def _redirect_logging(self):
...
+        # self.log_file = Path(os.getcwd()) / self._LOG_NAME
         # Remove log file if present
-        if os.path.exists(self.log_file):
-            os.remove(self.log_file)
+        # if os.path.exists(self.log_file):
+        #     os.remove(self.log_file)
 
         # Initiate logger object
         self.logger = logging.getLogger("SPACE")
         self.logger.setLevel(logging.DEBUG)
```
Comment (wsp-mbuckley) [inline src/caf/space/ui.py:818]: Get the `__package__` logger to be consistent

```suggestion
        self.logger = logging.getLogger(__package__)
        self.logger.setLevel(logging.DEBUG)
```

File: src/caf/space/ui.py [hunk 4]
```diff
@@ -798,38 +797,38 @@ def _redirect_logging(self):
         self.console_handler.setFormatter(fmt)
```
Comment (wsp-mbuckley) [inline src/caf/space/ui.py:797]: Use `__package__` instead of `"SPACE"` on line 794 above this

**Review by wsp-mbuckley — CHANGES_REQUESTED** [review]
Looks like you've implemented LogHelper correctly, the only thing is I would use `__package__` for the root logger instead of `"SPACE"`, this would mean you can use `LOG = logging.getLogger(__name__)` in all the modules like you have used in 'inputs.py'.

Also, when you're in the `NotebookApp._redirect_logging` function you're adding to the root logger but not to the Python warnings logger ("py.warnings") which means Python warnings won't show up in your GUI but will show up in the console window and the log file.

I think a new `add_handler` method should be added to `LogHelper` in `caf.toolkit`, which adds handlers to the root logger and warning logger and can be used by other people.

**Comment by isaac-tfn** [comment]
@wsp-mbuckley I only just remembered the suggested changes in this review, could you take a quick 5 min look and (hopefully) approve?

**Comment by wsp-mbuckley** [comment]
> @wsp-mbuckley I only just remembered the suggested changes in this review, could you take a quick 5 min look and (hopefully) approve?

There were 3 comments where I said use `__package__` instead of "SPACE" for the `getLogger` functions which haven't been changed yet, please could you do those. I think everything else looks good

**Comment by isaac-tfn** [comment]
> > @wsp-mbuckley I only just remembered the suggested changes in this review, could you take a quick 5 min look and (hopefully) approve?
> 
> There were 3 comments where I said use `__package__` instead of "SPACE" for the `getLogger` functions which haven't been changed yet, please could you do those. I think everything else looks good

I thought I'd changed them - I'll make sure I've pushed all my changes sorry

**Comment by isaac-tfn** [comment]
@wsp-mbuckley I might be missing something but I see 4 places you've requested "SPACE" changes to __package__ and I've made those changes. All of the instances seem to be in 'ui' and '__main__' so if there are more places it needs changing I don't think I can see those comments

**Review by wsp-mbuckley — APPROVED** [review]

**Comment by isaac-tfn** [comment]
> > @wsp-mbuckley I might be missing something but I see 4 places you've requested "SPACE" changes to **package** and I've made those changes. All of the instances seem to be in 'ui' and '**main**' so if there are more places it needs changing I don't think I can see those comments
> 
> Sorry Isaac, you're right when I looked at the comments in VS code it linked back to the state of the file at the commit when I made the comment (I guess trying to be helpful?)

No problem - I think suggestions is relatively new? (or you and Ben have both found the feature at about the same time) so maybe not quite optimised in VS code. Thanks for the review will merge now.
